sdk/event/evplugin/event_plugin.go

```diff
@@ -0,0 +1,32 @@
+package evplugin
+
+import "context"
+
+// these are just stubs for now
+
+// EventPlugin allows plugins to enable new ways to subscribe to events.
+// Plugins should implement the server and Vault should implement the client.
+type EventPlugin interface {
+	// Initialize the plugin.
+	Initialize(ctx context.Context, req InitializeRequest) (InitializeResponse, error)
+	// Subscribe creates a new subscription. Only a single subscription will be created per plugin.
+	Subscribe(ctx context.Context, req SubscribeRequest) (SubscribeResponse, error)
+	// ReceiveEvents is called with a stream of events for a single subscription.
+	ReceiveEvents(ctx context.Context, subscriptionID uint32, events <-chan SubscriptionEvent) (ReceiveEventsResponse, error)
```

Comment:
Combining might be the right call. I didn't use `SendEvents` because I thought it might be confusing since there is also the `logical.EventSender` interface that the plugin uses.